interfaces/builtin/repowerd.go

```diff
@@ -0,0 +1,335 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2016 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package builtin
+
+import (
+	"github.com/snapcore/snapd/interfaces"
+)
+
+var repowerdPermanentSlotAppArmor = []byte(`
+# Description: Allow operating as the repowerd service. Reserved because this
+#  gives privileged access to the system.
+# Usage: reserved
+
+# HW alarms
+/dev/alarm rw,
+
+# Suspend
+/sys/power/** rw,
+
+# Backlight
+/sys/**/brightness rw,
```

Comment:
This is a rough spot in our implementation and exists elsewhere (eg udev with /sys and /run/udev). For a slot implementation, it is not ideal to have a glob rule, but it also isn't the worst thing, so I'm ok with this type of rule if it can be scoped some. `/sys/**/type r,` (below) is pretty icky for example. Granted, it is read only, but there are a *lot* of files named 'type' in /sys (kernel, usb, firmware, etc).

Would this work:
```
# Allow adjusting backlights
# TODO: fine-tune this to resolve symlinks to use fine-grained rules
￼/sys/devices/{platform,pci*}/**/brightness rw,
/sys/devices/{platform,pci*}/**/max_brightness r,
/sys/devices/{platform,pci*}/**/type r,
```

One way to avoid this would be to declare to snapd what brightness files we are interested in (eg, /sys/class/backlight/* and /sys/class/leds/lcd-backlight/*), then have snapd resolve the symlinks to absolute paths, then create a sorted list of apparmor rules to put in the policy. This would work fine for backlights because those won't be hotpluggable. It also is cheap for snapd to do as part of its state operations and because the list is sorted, the policy shouldn't ever change so there is no apparmor parser overhead with this approach.